docs/mllib-data-types.md

```diff
@@ -226,7 +226,8 @@ examples = MLUtils.loadLibSVMFile(sc, "data/mllib/sample_libsvm_data.txt")
 
 A local matrix has integer-typed row and column indices and double-typed values, stored on a single
 machine.  MLlib supports dense matrices, whose entry values are stored in a single double array in
-column major.  For example, the following matrix `\[ \begin{pmatrix}
+column-major order, and sparse matrices, whose non-zero entry values are stored in the Compressed Sparse
+Column (CSC) format in column-major order.  For example, the following dense matrix `\[ \begin{pmatrix}
```

Comment:
Well I think documenting that we are using CSC representation gives the user some reasoning for how the `colPtrs`, `rowIndices`, and `values` parameters interact, and how to use them properly.  Now whether we should *require* that only those parameters be used, as opposed to `values` & `(row, col)` tuples, for example, is another question, and may be worth exploring.